File: src/main/java/net/nodus/database/site/SiteKeyRepository.java
package net.nodus.database.site;

import java.time.Instant;
import java.util.List;
import java.util.Optional;
import net.nodus.database.site.entity.SiteKey;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface SiteKeyRepository extends MongoRepository<SiteKey, String> {

    Optional<SiteKey> findBySiteIdAndStatusAndDeletedAtIsNull(String siteId, SiteKeyStatus status);

    List<SiteKey> findAllByKeyPrefixAndStatusAndDeletedAtIsNull(String keyPrefix,
        SiteKeyStatus status);

    List<SiteKey> findByDeletedAtBefore(Instant deletedAt);
}

File: src/main/kotlin/net/nodus/site/SiteKeyRepository.kt
package net.nodus.site

import net.nodus.site.entity.SiteKey
import org.springframework.data.mongodb.repository.MongoRepository
import java.time.Instant

interface SiteKeyRepository : MongoRepository<SiteKey, String> {
    fun findBySiteIdAndStatusAndDeletedAtIsNull(siteId: String, status: SiteKeyStatus): SiteKey?
    fun findAllByKeyPrefixAndStatusAndDeletedAtIsNull(keyPrefix: String, status: SiteKeyStatus): List<SiteKey>

    fun findByDeletedAtBefore(deletedAt: Instant): List<SiteKey>
}